parse.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 var (
-	versionRegexp           = regexp.MustCompile(`##? (?i:(HEAD|v?\d+.\d+(.\d+)?)( / (\d{4}-\d{2}-\d{2}))?)`)
+	versionRegexp           = regexp.MustCompile(`##? (?i:(HEAD|v?\d+\.\d+(?:\.\d+)?))(?U:.*)(\d{4}-\d{2}-\d{2})?.?$`)
```

Comment:
What does `?U` do here?

Reply:
According to https://pkg.go.dev/regexp/syntax, the `?` starts a section of flags, and `U` sets the flag `ungreedy: swap meaning of x* and x*?, x+ and x+?, etc`.